tokio/src/task/task_local.rs

```diff
@@ -332,6 +332,76 @@ pin_project! {
     }
 }
 
+impl<T, F> TaskLocalFuture<T, F>
+where
+    T: 'static,
+{
+    /// Takes the task local value `T` owned by the `TaskLocalFuture`. If the
+    /// task local value exists, then returns `Some(T)` and the task local value
+    /// inside the `TaskLocalFuture` becomes unset. If it does not exist,
+    /// it returns `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # async fn dox() {
+    ///  tokio::task_local! {
+    ///     static KEY: u32;
+    ///  }
+    ///
+    ///  let fut = KEY.scope(42, async {
+    ///     // Do some async work
+    ///  });
+    ///
+    ///  let mut pinned = Box::pin(fut);
+    ///
+    ///  // Complete the TaskLocalFuture
+    ///  let _ = (&mut pinned).as_mut().await;
```

Comment:
I would think that the `.as_mut()` is enough to make this compile.
```suggestion
    ///  // Complete the TaskLocalFuture
    ///  let _ = pinned.as_mut().await;
```